unyt/dask_array.py

```diff
@@ -0,0 +1,620 @@
+
+        if hasattr(unyt_result, "units"):
+            return _create_with_quantity(daskresult, unyt_result)
+        return daskresult  # __lt__, __le__, etc. will hit this
+
+    return wrapper
+
+
+class unyt_dask_array(_dask_Array):
+    """
+    a dask.array.core.Array subclass that tracks units. Easiest to use the
+    unyt_from_dask helper function to generate new instances.
+
+    Parameters
+    ----------
+
+    All parameters are those for dask.array.core.Array
+
+    """
+
+    def __new__(
+        clss,
+        dask_graph,
+        name,
+        chunks,
+        dtype=None,
+        meta=None,
+        shape=None,
+        units=None,
+        registry=None,
+        bypass_validation=False,
+        input_units=None,
+        unyt_name=None,
+    ):
+
+        # get the base dask array
+        obj = super(unyt_dask_array, clss).__new__(
+            clss,
+            dask_graph,
+            name,
+            chunks,
+            dtype,
+            meta,
+            shape,
+        )
+
+        # attach our unyt sidecar quantity
+        dtype = obj.dtype
+        obj._unyt_array = ua.unyt_array(
+            [1.0],
+            units,
+            registry,
+            dtype,
+            bypass_validation,
+            input_units,
+            unyt_name,
+        )
+
+        obj.units = obj._unyt_array.units
+        obj.unyt_name = obj._unyt_array.name
+
+        # set the unit conversion attributes so they are discoverable. no name
+        # conflicts for now, but this could be an issue if _unyt_funcs_to_track
+        # is expanded.
+        for attr in _unyt_funcs_to_track:
+            setattr(obj, attr, getattr(obj._unyt_array, attr))
+
+        return obj
+
+    def _elemwise(self, ufunc, *args, **kwargs):
+        args, unyt_result = _prep_ufunc(ufunc, *args, **kwargs)
+        return _post_ufunc(super()._elemwise, unyt_result)(ufunc, *args, **kwargs)
+
+    def __array_ufunc__(self, numpy_ufunc, method, *inputs, **kwargs):
+        inputs, unyt_result = _prep_ufunc(
+            numpy_ufunc, *inputs, extract_dask=True, **kwargs
+        )
+        wrapped_func = _post_ufunc(super().__array_ufunc__, unyt_result)
+        return wrapped_func(numpy_ufunc, method, *inputs, **kwargs)
+
+    def __array_function__(self, func, types, args, kwargs):
+        args, unyt_result = _prep_ufunc(func, *args, extract_dask=True, **kwargs)
+        types = [type(i) for i in args]
+        wrapped_func = _post_ufunc(super().__array_function__, unyt_result)
+        return wrapped_func(func, types, args, kwargs)
+
+    def __getitem__(self, index):
+        return _simple_dask_decorator(super().__getitem__, self)(index)
+
+    def __setitem__(self, key, value):
+        # requires dask >= 2021.4.1
+        super().__setitem__(key, value)
+
+    def __repr__(self):
+        disp_str = super().__repr__().replace("dask.array", "unyt_dask_array")
+        units_str = f", units={self.units.__str__()}>"
```

Comment:
I don't think this makes a difference ?
```suggestion
        units_str = f", units={self.units}>"
```